src/mpid/ch4/netmod/ofi/ofi_capability_sets.h

```diff
@@ -0,0 +1,198 @@
+MPL_STATIC_INLINE_PREFIX int MPIDI_OFI_get_set_number(char *set_name)
+{
+    if (!strcmp("psm", set_name)) {
+        return 0;
+    } else if (!strcmp("psm2", set_name)) {
+        return 1;
+    } else if (!strcmp("gni", set_name)) {
+        return 2;
+    } else if (!strcmp("sockets", set_name)) {
+        return 3;
+    } else if (!strcmp("bgq", set_name)) {
+        return 4;
+    } else {
+        return -1;
+    }
+}
+
+/*
+ * The definitions map to these capability sets:
+ *
+ * MPIDI_OFI_ENABLE_DATA               fi_tsenddata (and other functions with immediate data)
+ *                                     Uses FI_REMOTE_CQ_DATA, FI_DIRECTED_RECV
+ * MPIDI_OFI_ENABLE_AV_TABLE           Use FI_AV_TABLE instead of FI_AV_MAP
+ * MPIDI_OFI_ENABLE_SCALABLE_ENDPOINTS fi_scalable_ep instead of fi_ep
+ *                                     domain_attr.max_ep_tx_ctx > 1
+ * MPIDI_OFI_ENABLE_STX_RMA            Use shared transmit contexts for RMA
+ *                                     Uses FI_SHARED_CONTEXT
+ * MPIDI_OFI_ENABLE_MR_SCALABLE        Use FI_MR_SCALABLE instead of FI_MR_BASIC
+ *                                     If using runtime mode, this will be set to FI_MR_UNSPEC
+ * MPIDI_OFI_ENABLE_TAGGED             Use FI_TAGGED interface instead of FI_MSG
+ * MPIDI_OFI_ENABLE_AM                 Use FI_MSG and FI_MULTI_RECV for active messages
+ * MPIDI_OFI_ENABLE_RMA                Use FI_ATOMICS and FI_RMA interfaces
+ */
+
+#define MPIDI_OFI_ENABLE_DATA_PSM               MPIDI_OFI_OFF
+#define MPIDI_OFI_ENABLE_AV_TABLE_PSM           MPIDI_OFI_ON
+#define MPIDI_OFI_ENABLE_SCALABLE_ENDPOINTS_PSM MPIDI_OFI_OFF
+#define MPIDI_OFI_MAX_ENDPOINTS_PSM             MPIDI_OFI_MAX_ENDPOINTS_REGULAR
+#define MPIDI_OFI_MAX_ENDPOINTS_BITS_PSM        MPIDI_OFI_MAX_ENDPOINTS_BITS_REGULAR
+#define MPIDI_OFI_ENABLE_STX_RMA_PSM            MPIDI_OFI_OFF
+#define MPIDI_OFI_ENABLE_MR_SCALABLE_PSM        MPIDI_OFI_ON
+#define MPIDI_OFI_ENABLE_TAGGED_PSM             MPIDI_OFI_ON
+#define MPIDI_OFI_ENABLE_AM_PSM                 MPIDI_OFI_ON
+#define MPIDI_OFI_ENABLE_RMA_PSM                MPIDI_OFI_ON
+
+#ifdef MPIDI_CH4_OFI_USE_SET_PSM
+#define MPIDI_OFI_ENABLE_RUNTIME_CHECKS     MPIDI_OFI_OFF
+#define MPIDI_OFI_ENABLE_DATA               MPIDI_OFI_ENABLE_DATA_PSM
+#define MPIDI_OFI_ENABLE_AV_TABLE           MPIDI_OFI_ENABLE_AV_TABLE_PSM
+#define MPIDI_OFI_ENABLE_SCALABLE_ENDPOINTS MPIDI_OFI_ENABLE_SCALABLE_ENDPOINTS_PSM
+#define MPIDI_OFI_MAX_ENDPOINTS             MPIDI_OFI_MAX_ENDPOINTS_PSM
+#define MPIDI_OFI_MAX_ENDPOINTS_BITS        MPIDI_OFI_MAX_ENDPOINTS_BITS_PSM
+#define MPIDI_OFI_ENABLE_STX_RMA            MPIDI_OFI_ENABLE_STX_RMA_PSM
+#define MPIDI_OFI_ENABLE_MR_SCALABLE        MPIDI_OFI_ENABLE_MR_SCALABLE_PSM
+#define MPIDI_OFI_ENABLE_TAGGED             MPIDI_OFI_ENABLE_TAGGED_PSM
+#define MPIDI_OFI_ENABLE_AM                 MPIDI_OFI_ENABLE_AM_PSM
+#define MPIDI_OFI_ENABLE_RMA                MPIDI_OFI_ENABLE_RMA_PSM
+#endif
```

Comment:
@wesbland: I'm missing something here.  I'm not sure I understand how these extra variables helps with the CVARs.  I'm not suggesting you change the "runtime" versions of `MPIDI_OFI_ENABLE_DATA`, etc., but just the provider-specific versions.